lib/ansible/modules/windows/win_scheduled_task.ps1

```diff
@@ -295,51 +328,97 @@ Function Compare-PropertyList {
+                            }
+
+                            if ($sub_existing_value -ne $null) {
+                                [void]$diff_list.Add("-$sub_com_name=$sub_existing_value")
+                            }
+                        }
                     }
-                    if ($new_value -ne $null) {
-                        [void]$diff_list.Add("+$com_name=$new_value")
+                    else {
+                        if ($property_value -ne $null) {
+                            [void]$diff_list.Add("+$com_name=$property_value")
+                        }
+
+                        if ($existing_value -ne $null) {
+                            [void]$diff_list.Add("-$com_name=$existing_value")
+                        }
                     }
                 }
             }
 
             [void]$changes.Add("$property_name[$i] = {`n  $($diff_list -join ",`n  ")`n}")
         } else {
             # compare the properties of existing and new
-            $diff_list = [System.Collections.ArrayList]@()
-
-            foreach ($property_arg in $total_args) {
-                $new_value = $new_property.$property_arg
-                $com_name = Convert-SnakeToPascalCase -snake $property_arg
-                $existing_value = $existing_property.$com_name
+            $properties = $new_property.Keys | Where-Object { $total_args -contains $_ }
+            if ($properties -ne $null) {
+                $diff_list = [System.Collections.ArrayList]@()
 
-                if ($new_value -ne $null) {
-                    if ($new_value -cne $existing_value) {
-                        [void]$diff_list.Add("-$com_name=$existing_value")
-                        [void]$diff_list.Add("+$com_name=$new_value")
+                foreach ($property_arg in $properties) {
+                    $com_name = Convert-SnakeToPascalCase -snake $property_arg
+                    $property_value = $new_property.$property_arg
+                    $existing_value = $existing_property.$com_name
+                    
+                    if ($property_value -is [Hashtable]) {
+                        foreach ($sub_property_arg in $property_value.Keys) {
+                            $sub_property_value = $property_value.$sub_property_arg
+                            
+                            if ($sub_property_value -ne $null) {
+                                $sub_com_name = Convert-SnakeToPascalCase -snake $sub_property_arg
+                                $sub_existing_value = $existing_property.$com_name.$sub_com_name
+
+                                if ($sub_property_value -cne $sub_existing_value) {
+                                    [void]$diff_list.Add("-$sub_com_name=$sub_existing_value")
+                                    [void]$diff_list.Add("+$sub_com_name=$sub_property_value")
+                                }
+                            }
+                        }
+                    }
+                    else {
```

Comment:
Instead of the nested if statements this probably could be


```
} else if ($property_value -ne $null -and $property_value -cne $existing_value) {
    [void]$diff_list....
    ...
}
```